quadtree.js

```diff
@@ -26,13 +26,85 @@ class Rectangle {
       point.y <= this.y + this.h);
   }
 
-  intersects(range) {
-    return !(range.x - range.w > this.x + this.w ||
-      range.x + range.w < this.x - this.w ||
-      range.y - range.h > this.y + this.h ||
-      range.y + range.h < this.y - this.h);
-  }
+    intersects(range) {
+
+      // get the name of the range's class (i.e Rectangle, Circle etc.)
+      var is_Rect = range.constructor.name;
+
+      // if the range is of class Rectangle
+      if(is_Rect == 'Rectangle'){
+        return !(range.x - range.w > this.x + this.w ||
+                 range.x + range.w < this.x - this.w ||
+                 range.y - range.h > this.y + this.h ||
+                 range.y + range.h < this.y - this.h);
+      }else {
+          // if the range is of class Circle
+          if(is_Rect == 'Circle'){
+
+              // call the intersects function of the range
+              // this is done in order to have each class
+              // having its own intersects and not have here
+              // a huge function that contains all of the
+              // intersection checks
+              return range.intersects(this);
+          }
+      }
+    }
+
+    show(){
+        rect(this.x, this.y, this.w * 2, this.h * 2);
```

Comment:
It has been said to keep the library free of p5.js

Reply:
yes I know, but this was in the original code :)

I just moved it to the show method

Reply:
The library is only the quadtree.js file, the show function was in the sketch.js (not part of the library)

Reply:
Iv'e commited the changes, is it ok now?